CRM/ACL/BAO/ACL.php

```diff
@@ -862,6 +862,13 @@ public static function group(
     $allGroups = NULL,
     $includedGroups = NULL
   ) {
+    $userCacheKey = "{$contactID}_{$type}_{$tableName}_" . CRM_Core_Config::domainID() . '_' . md5(implode(',', array_merge((array) $allGroups, (array) $includedGroups)));
+    if (empty(Civi::$statics[__CLASS__]['permissioned_groups'])) {
+      Civi::$statics[__CLASS__]['permissioned_groups'] = array();
+    }
+    if (!empty(Civi::$statics[__CLASS__]['permissioned_groups'][$userCacheKey])) {
+      return Civi::$statics[__CLASS__]['permissioned_groups'][$userCacheKey];
+    }
```

Comment:
Its just am afraid of any unintended regression when  the second parameter ```$contactID``` is empty so in this case the  ```$userCacheKey``` will be built without the contactID and eventually only $ids (w/ hidden group) will be assigned to it. Anyways its ok to ignore for now.